__tests__/app.test.js

```diff
@@ -142,6 +143,71 @@ describe('GET /api/articles/:article_id', () => {
       });
   });
 });
+
+describe('GET /api/articles/:article_id/comments', () => {
+  test('returns array of 11 comments for  article_id 1', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body;
+        expect(comments.length).toBe(11);
+      });
+  });
+  test('returns array of 0 comments for  article_id 2', () => {
+    return request(app)
+      .get('/api/articles/2/comments')
+      .then((response) => {
+        const { comments } = response.body;
+        expect(comments.length).toBe(0);
+      });
+  });
+  test('each comment object has requirest keys', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .then((response) => {
+        const { comments } = response.body;
+        comments.forEach((comment) =>
+          expect(comment).toEqual(
+            expect.objectContaining({
+              comment_id: expect.any(Number),
+              body: expect.any(String),
+              article_id: expect.any(Number),
+              author: expect.any(String),
+              votes: expect.any(Number),
+              created_at: expect.any(String),
+            })
+          )
+        );
+      });
+  });
+  test('comments should be sorted by the most recent one', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .then((response) => {
+        const { comments } = response.body;
+        expect(comments).toBeSortedBy('created_at', {
+          descending: true,
+        });
+      });
+  });
+  test('returns an error if article id has not been found', () => {
+    return request(app)
+      .get('/api/articles/99999999/comments')
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe('Not found');
+      });
+  });
+  test('returns an error if article id is not an int', () => {
+    return request(app)
+      .get('/api/articles/BOOM/comments')
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe('Bad request');
+      });
+  });
```

Comment:
Your tests look robust and include error handling. It is nice to see that you have checked for the length of the array and you have checked that the comments are sorted correctly.